src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -184,7 +184,27 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
 
   override def clone() = new ArrayDeque(array.clone, start, end)
 
-  override def clear() = if (nonEmpty) set(array = ArrayDeque.alloc(ArrayDeque.DefaultInitialSize), start = 0, end = 0)
+  /**
+    * Note: This does not actually resize the internal representation.
+    * See clearAndShrink if you want to also resize internally
+    */
+  override def clear() = {
+    nullify()
+    start = 0
+    end = 0
+  }
```

Comment:
This is now inconsistent with the auto-shrinking, given my earlier example of clearing, then adding 101, then removing 1, etc.

I feel like there are two options for consistency sake:

Behavior 1:  no auto-shrink on removals, no shrinking on clear
Behavior 2: auto-shrink on removals, shrink on clear

Either this is represented in types `ArrayDeque` + `ArrayDeque with ManualShrink`  or a subclass; or there is extra internal state to track if there has been a size request.   I'm not personally a fan of the extra internal state but others might not be a fan of an extra type.    To minimize differences, both can have `clearNoShrink` and `clearAndShrink` but one implements `clear` with one and the other type implements its `clear` with the other.  Additionally, the two types would need to differ on whether there was any auto-shrinking during removals.

What are other opinions on this?   I see a few more use cases now where auto-shrink is useful -- I was thinking along the lines of when I would use a Buffer, but Queue and Deque use cases that I can imagine have more situations where auto-shrink would be useful.  I tend to think of a Buffer as an intermediate data structure during transformation of other data structures.  Whereas a Queue and Deque are often longer lived.